### PR TITLE
Feature/custom columns color

### PR DESCRIPTION
**Pull request by j8seangel**

### CODE REVIEW

File: app/assets/javascripts/views/map_file_categories_view.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          self.updateColumnsColor(color);
+        });
+      });
+
+    },
+
+    getCategories: function(table, column) {
+      var query = 'SELECT DISTINCT ' + column +' AS CATEGORY FROM ' + table + ' LIMIT 15';
+      var defer = new $.Deferred();
+      $.getJSON('https://'+this.data.cartodb_user+'.cartodb.com/api/v2/sql/?q='+query)
```
Comment (geriux) [inline app/assets/javascripts/views/map_file_categories_view.js]: I saw that CartoDB has an example with getJSON, but just so you know CartoDB.js also has a method to make SQL queries http://docs.cartodb.com/cartodb-platform/cartodb-js/sql/#arguments

**Comment by geriux** [comment]
👏 great work!! Waiting for the pending fixes and testing to merge!

**Comment by j8seangel** [comment]
Done, and thanks 😁 !

File: app/assets/javascripts/views/map_file_columns_view.js [hunk 2]
```diff
@@ -158,17 +170,24 @@
...
       }
       this.fileInput.value = '';
-      this.inputWrapper.classList.remove('_hidden');
+      this.el.getElementsByClassName('input-wrapper')[0].classList.remove('_hidden');
     },
 
-    onAddItem: function(e) {
-      e.currentTarget.classList.add('_hidden');
+    onAddNewItem: function(e) {
+      debugger;
```
Comment (geriux) [inline app/assets/javascripts/views/map_file_columns_view.js]: 👀

**Pull request merged**